### ISSUE TITLE
Browse Class in code pane only works if entire class selected (senders/implementors correctly looks at AST)

### ISSUE THREAD

**Issue by timm-mwp** [guillep/pharo-core#9615]
**Bug description**
When browsing code in the system browser, pressing cmd-b with your cursor on a class gives an error "variable or expression expected", whereas it should look at the AST of the method and understand you are on a class (this is what sender/implementors does)

**To Reproduce**
1. Go to a method in a SystemBrowser that has a class
2. Click on a class name (that has other code on the same line)
3. Press cmd-b with your cursor sitting inside the class name
4. See error " Variable or expression expected...."

**Expected behavior**
The command should understand you are sitting on a class name (by looking at the AST) and browse the class (the fallback should be to see if you have selected something)

**Screenshots**
![image](https://user-images.githubusercontent.com/65404874/126283028-5175e154-0c8b-4a1f-b3d4-3df339c778ba.png)
![image](https://user-images.githubusercontent.com/65404874/126283069-3d583de9-558d-4f8e-84af-ed11bba4852c.png)

**Version information:**
 - Version: [e.g. 1.6.1]
 - Pharo Version [e.g 9] 9

**Expected development cost**
2 hours - look at how senders/implementors code works

**Additional context**

**Comment by MarcusDenker** [comment]
yes, we need to fix this. Right now it just selects the line and evaluates, that is not good, interaction is one thing, but the problem is that evaluation leads to many problems:

- put a cursor in the "Smalltalk of "Smalltalk snapshot: true andQuit:  true", browse it --> image quits  https://github.com/pharo-project/pharo/issues/8515
- select parts of a class, browse it --> broken dialog to define variables pop up https://github.com/pharo-project/pharo/issues/9644
- browse it of selectors does not work: https://github.com/pharo-project/pharo/issues/8889

**Comment by timm-mwp** [comment]
somewhat related - if done right, it could mean that Browse, and Browse Implementors end up being the same thing? If we infer what your cursor is on (e.g. the closest AST node) - then browse should show you what that thing is - for a Class, show the class; for a method, show the method definition; not sure what to do for a symbol (ideally show where its defined if we can - but this could be a later refinement), for an ivar also show where its defined in the current scope? (again a more advanced idea).

Given the above, I can't see how Browse Implementers is then any different from Browse (we wouldn't need the distinction). So we could point them to the same code and could potentially even deprecate it and free up a keystroke/complexity. 

Browse senders is of course a bit different - I don't think we could deduce that you wanted to do this specific action, and so we would need a separate menu/keystroke for that (unless the UI for a Browse, actually did both simultaneously - which could be neat/useful - but its probably a v2 sort of step)
